### ISSUE TITLE
[parse5-utils] isHtmlFragment doesn't match documents starting with comment

### ISSUE THREAD

**Issue by pintomic** [modernweb-dev/web#1110]
Consider this sample html:

```html
<!--  -->

<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <meta http-equiv="X-UA-Compatible" content="ie=edge" />

    <title>MyApplication</title>
    <meta name="description" content="MyApplication description" />

  </head>
  <body>
    <app-index></app-index>

    <noscript>Please enable JavaScript to view this website.</noscript>

    <script type="module" src="src/components/app-index.ts"></script>
  </body>
</html>
```

[isHtmlFragment](https://github.com/modernweb-dev/web/blob/0cc777708312833adbcac7ab9f03ca865ff3bafb/packages/parse5-utils/src/index.js#L49-L51) would return `false` since the regex used doesn't allow comments as first part of the document.

**Comment by SBylemans** [comment]
When your appIndex file starts with a comment, this has a rather annoying side effect with regards to the watchplugin. 

The plugin is not able to append the websocket script to the appIndex file because before attaching this script it uses the isHtmlFragment check.

So using `watch:true` with an appIndex file that starts with a comment, does not actually trigger the reload when making changes to a file in the filesystem.

**Comment by abdonrd** [comment]
There are any news about this? Thanks in advance!

**Comment by SBylemans** [comment]
I created a PR: https://github.com/modernweb-dev/web/pull/1284

**Comment by web-padawan** [comment]
Fixed by #1284 and #1288
